tests/30rooms/11leaving.pl

```diff
@@ -0,0 +1,292 @@
+
+            my $madeup_test_state =
+                first { $_->{type} eq "madeup.test.state" } @{$room->{state}};
+
+            $madeup_test_state->{content}{body}
+                eq "S1. B's state before A left"
+                or die "Received state that happened after leaving the room";
+
+            $room->{messages}{chunk}[0]{content}{body}
+                eq "M2. B's message before A left"
+                or die "Received message that happened after leaving the room";
+
+            Future->done(1);
+        })
+    };
+
+test "Can get rooms/{roomId}/initialSync for a departed room (SPEC-216)",
+    requires => [qw( do_request_json )],
+
+    check => sub {
+        my ( $do_request_json ) = @_;
+
+        $do_request_json->(
+            method => "GET",
+            uri => "/api/v1/rooms/$room_id/initialSync",
+            params => { limit => 2 },
+        )->then( sub {
+            my ( $room ) = @_;
+
+            require_json_keys( $room, qw( state messages membership ) );
+
+            $room->{membership} eq "leave" or die "Membership is not leave";
+
+            my $madeup_test_state =
+                first { $_->{type} eq "madeup.test.state" } @{$room->{state}};
+
+            $madeup_test_state->{content}{body} eq "S1. B's state before A left"
+                or die "Received state that happened after leaving the room";
+
+            $room->{messages}{chunk}[0]{content}{body}
+                eq "M2. B's message before A left"
+                or die "Received message that happened after leaving the room";
+
+            not @{$room->{presence}}
+                or die "Received presence information after leaving the room";
+
+            not @{$room->{receipts}}
+                or die "Received receipts after leaving the room";
+
+            Future->done(1);
+        })
+    };
+
+test "Can get rooms/{roomId}/state for a departed room (SPEC-216)",
+    requires => [qw( do_request_json )],
+
+    check => sub {
+        my ( $do_request_json ) = @_;
+
+        $do_request_json->(
+            method => "GET",
+            uri => "/api/v1/rooms/$room_id/state",
+        )->then( sub {
+            my ( $state ) = @_;
+
+            my $madeup_test_state =
+                first { $_->{type} eq "madeup.test.state" } @{$state};
+
+            $madeup_test_state->{content}{body}
+                eq "S1. B's state before A left"
+                or die "Received state that happened after leaving the room";
+
+            Future->done(1);
+        })
+    };
+
+test "Can get rooms/{roomId}/members for a departed room (SPEC-216)",
+    requires => [qw( do_request_json )],
+
+    check => sub {
+        my ( $do_request_json ) = @_;
+
+        $do_request_json->(
+            method => "GET",
+            uri => "/api/v1/rooms/$room_id/members",
+        )->then( sub {
+            my ( $body ) = @_;
+
+            require_json_keys( $body, qw( chunk ) );
```

Comment:
No assertions on its contents?

(Specifically, that the left user is still in the member list)